procrastinate/sql/migrations/02.00.03_01_add_job_priority.sql

```diff
@@ -0,0 +1,122 @@
+ALTER TABLE procrastinate_jobs ADD COLUMN priority integer DEFAULT 0 NOT NULL;
+
+DROP FUNCTION IF EXISTS procrastinate_defer_job(character varying, character varying, text, text, jsonb, timestamp with time zone);
+CREATE OR REPLACE FUNCTION procrastinate_defer_job(
+    queue_name character varying,
+    task_name character varying,
+    priority integer,
+    lock text,
+    queueing_lock text,
+    args jsonb,
+    scheduled_at timestamp with time zone
+)
+    RETURNS bigint
+    LANGUAGE plpgsql
+AS $$
+DECLARE
+	job_id bigint;
+BEGIN
+    INSERT INTO procrastinate_jobs (queue_name, task_name, priority, lock, queueing_lock, args, scheduled_at)
+    VALUES (queue_name, task_name, priority, lock, queueing_lock, args, scheduled_at)
+    RETURNING id INTO job_id;
+
+    RETURN job_id;
+END;
+$$;
+
+DROP FUNCTION IF EXISTS procrastinate_defer_periodic_job(character varying, character varying, character varying, character varying, character varying, bigint, jsonb);
+CREATE OR REPLACE FUNCTION procrastinate_defer_periodic_job(
```

Comment:
I think it would make sense if the priority was customizable on periodic jobs, but we don't have to do that on this PR. If you'd rather not, let's create an issue so we don't forget.

Reply:
I also considered that, but I was not sure if it was really necessary or what the API would look like. Let's create a new issue for that; this one is large enough now 😉 But I can work on that, too.

Reply:
Tracked in #1072